vuvuzela_test.go

```diff
@@ -0,0 +1,619 @@
+	}
+	if !bytes.Equal(aliceMsgs[0].EncryptedMessage[:], bobReplies[0][:]) {
+		t.Fatal("alice's message should be the same as bob's reply")
+	}
+
+	latestRound, err := aliceClient.LatestRound()
+	if err != nil {
+		t.Fatalf("Failed to get latest round: %s", err)
+	}
+	if bobRound > latestRound || aliceRound > latestRound || aliceReplyRound > latestRound || bobReplyRound > latestRound {
+		t.Fatalf("Latest round %d should be same or later than %d, %d %d, %d", latestRound, bobRound, aliceRound, aliceReplyRound, bobReplyRound)
+	}
+}
+
+func TestVuvuzelaConfig(t *testing.T) {
+	u := createVuvuzelaUniverse()
+	defer func() {
+		// Give time for goroutines to finish before pulling the rug out from under them
+		time.Sleep(1 * time.Second)
+		u.Destroy()
+	}()
+
+	user := u.createUser("user")
+	_, err := user.ConnectConvo()
+	if err != nil {
+		t.Fatalf("Failed to connect user's conversation: %s", err)
+	}
+	defer user.CloseConvo()
+
+	// Deep copy previous convo config
+	data, err := json.Marshal(u.ConvoConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	newConvoConfig := new(config.SignedConfig)
+	err = json.Unmarshal(data, newConvoConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	newConvoConfig.Created = time.Now()
+	newConvoConfig.Expires = time.Now().Add(24 * time.Hour)
+	newConvoConfig.PrevConfigHash = u.ConvoConfig.Hash()
+
+	newMixchain := mock.LaunchMixchain(2, u.CoordinatorPub)
+	newConvoConfig.Inner.(*convo.ConvoConfig).MixServers = newMixchain.Servers
+
+	sig := ed25519.Sign(u.GuardianPriv, newConvoConfig.SigningMessage())
+	newConvoConfig.Signatures = map[string][]byte{
+		base32.EncodeToString(u.GuardianPub): sig,
+	}
+
+	err = u.ConfigClient.SetCurrentConfig(newConvoConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	confs := <-user.Handler.(*mockHandler).newConfig
+	if confs[0].Hash() != newConvoConfig.Hash() {
+		t.Fatalf("received unexpected config: %s", debug.Pretty(confs))
+	}
+}
+
+func TestRunVuvuzela(t *testing.T) {
+	// t.Skip("skipping test that runs vuvuzela servers forever")
+
+	u := createVuvuzelaUniverse()
+	u.createAlpenhornUniverse()
+
+	client := u.createAlpenhornClient("alice@example.org")
+
+	_, err := client.ConnectAddFriend()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = client.ConnectDialing()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(2 * time.Second)
+
+	_, err = client.SendFriendRequest("alice@example.org", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	<-client.Handler.(*alpenhornClientHandler).sentFriendRequest
+	log.Info("Sent friend request")
+
+	<-client.Handler.(*alpenhornClientHandler).confirmedFriend
+	log.Info("Confirmed friend request")
+
+	// Persist new friend
+	err = client.Persist()
+	if err != nil {
+		log.Panicf("persisting alpenhorn client: %s", err)
+	}
+
+	client.CloseAddFriend()
+	client.CloseDialing()
+
+	sigchan := make(chan os.Signal, 1)
+	signal.Notify(sigchan, os.Interrupt)
+
+	log.Infof("Running Vuvuzela locally with config servers at %s...", u.ConfigClient.ConfigServerURL)
```

Comment:
It's very hard to find this log message among all the others. Is there a better way I can display this so clients know where to connect?